### PR TITLE
Give explicit type for empty dictionary to fix Xcode 14.3 compile error

### PR DESCRIPTION
**Pull request by dingyishen-amazon**
## ℹ️ Description
*provide a summary of the changes and the related issue, relevant motivation and context*
After upgrading to Xcode 14.3, there are compiling issues caused by no explicit type for empty dictionary.
Fix by giving explicit type.

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
    - [ ] README update
    - [ ] CHANGELOG update
    - [ ] guides update
- [ ] This change requires a dependency update
    - [ ] Amazon Chime SDK Media
    - [ ] Other (update corresponding legal documents)

## 🧪 How Has This Been Tested?
*describe the tests that you ran to verify your changes, any relevant details for your test configuration*

### Additional Manual Test
- [ ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Rotate screen back and forth

## 📱 Screenshots, if available
*provide screenshots/video record if there's a UI change in demo app*

#### Integration validation (required before release)

- [ ] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [ ] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dylonChime — APPROVED** [review]

**Comment by hokyungh** [comment]
Will it break on old Xcode 14?

**Comment by dingyishen-amazon** [comment]
> Will it break on old Xcode 14?
No, have tested with old Xcode 14.

**Review by hokyungh — APPROVED** [review]

File: AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/backgroundfilter/BackgroundFilterProcessor.swift [hunk 1]
```diff
@@ -192,7 +192,7 @@ public class BackgroundFilterProcessor {
         attributes[kCVPixelBufferPixelFormatTypeKey] = NSNumber(value: Int(kCVPixelFormatType_32BGRA))
         attributes[kCVPixelBufferWidthKey] = NSNumber(value: newWidth)
         attributes[kCVPixelBufferHeightKey] = NSNumber(value: newHeight)
-        attributes[kCVPixelBufferIOSurfacePropertiesKey] = [:] as NSObject
+        attributes[kCVPixelBufferIOSurfacePropertiesKey] = [AnyHashable: Any]() as NSObject
```
Comment (georgezy-amzn) [inline AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/backgroundfilter/BackgroundFilterProcessor.swift:195]: Have you tried to build the demo? there are also similar issue in demo.

Comment (georgezy-amzn) [reply AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/backgroundfilter/BackgroundFilterProcessor.swift:195]: in CoreImageVideoProcessor.swift and MetalVideoProcessor.swift

Comment (dingyishen-amazon) [reply AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/backgroundfilter/BackgroundFilterProcessor.swift:195]: Yes, and I changed these 2 files as well.

**Review by georgezy-amzn — APPROVED** [review]

**Pull request merged**